<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Knoblauch Nicholas" />

<meta name="date" content="2017-03-02" />

<title>Results of first grid search</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-1.1/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-1.1/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">eqtl_estimation</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/workflowr">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Results of first grid search</h1>
<h4 class="author"><em>Knoblauch Nicholas</em></h4>
<h4 class="date"><em>2017-03-02</em></h4>

</div>


<p><strong>Last updated:</strong> 2017-03-29</p>
<p><strong>Code version:</strong> ac6247a</p>
<div id="introduction" class="section level2">
<h2>Introduction:</h2>
<p>This is the result of running RSSR on Whole Blood eQTL, using the 1 thousand genomes LD panel</p>
<pre class="r"><code>library(dplyr)
library(ggplot2)
library(tidyr)
library(RColumbo)
lnz_df &lt;- readRDS(&quot;../data/GTEx_rssr_results/Whole_Blood/lnzdf.RDS&quot;)
head(lnz_df) </code></pre>
<pre><code># A tibble: 6 × 7
  log10odds          lnZ  sigb chunk chrom fgeneid      tissue
      &lt;dbl&gt;        &lt;dbl&gt; &lt;dbl&gt; &lt;int&gt; &lt;int&gt;   &lt;dbl&gt;       &lt;chr&gt;
1 -4.999996    0.6784462 0.400     1     1  186470 Whole_Blood
2 -3.999957    2.2198926 0.400     1     1  186470 Whole_Blood
3 -2.999565   -4.3478718 0.400     1     1  186470 Whole_Blood
4 -2.298853  -51.8068613 0.400     1     1  186470 Whole_Blood
5 -1.995635 -116.8349443 0.400     1     1  186470 Whole_Blood
6 -4.999996    0.4294538 0.675     1     1  186470 Whole_Blood</code></pre>
<p>The <code>lnZ</code> column represents the log variational lower bound for the combination of <span class="math inline">\(\sigma_\beta\)</span> (<code>sigb</code>) and <span class="math inline">\(\theta=\log_{10}{\frac{\pi}{1-\pi}}\)</span> (<code>log10odds</code>) values used on chromosome <code>chrom</code> chunk number <code>chunk</code>. Chunks are at most 25000 SNPs. If there are <span class="math inline">\(P\)</span> chunks in the genome, then let’s call:</p>
<p><span class="math display">\[f_{p,g,i,j}(\theta_i,\sigma_{\beta,j},\hat{\beta}_{g,p},\hat{s}_{g,p})\]</span></p>
<p>the log-variational lower bound for gene <span class="math inline">\(g\)</span>, chunk <span class="math inline">\(p\)</span>, and grid entry <span class="math inline">\(i,j\)</span>.</p>
</div>
<div id="summarizing-for-one-gene" class="section level2">
<h2>Summarizing for one gene</h2>
<p>A key assumption of my model is that the chunks are independent. Let’s test that assumption by visualizing <span class="math inline">\(lnZ\)</span> across the chunks. This plot shows the sum of the variational lower bound (<code>lnZ</code>) for each chunk of each chromosome across the 25(5x5) combinations of parameter values. <span class="math display">\[ f_{p,g}(\hat{\beta}_g,\hat{s}_g)=\sum_{i=1}^5 \sum_{j=1}^5 f_{p,g,i,j}(\theta_i,{\sigma_{\beta}}_j,\hat{\beta}_{g,p},\hat{s}_{g,p})\]</span></p>
<pre class="r"><code>lnz_df &lt;- distinct(lnz_df,chunk,chrom) %&gt;% mutate(chunk_i=1:n()) %&gt;% group_by(chrom) %&gt;% mutate(chunk_p=(chunk-1)/sum(chunk)) %&gt;% ungroup() %&gt;% inner_join(lnz_df) %&gt;% mutate(chrom_chunk=chrom+chunk_p)</code></pre>
<pre><code>Joining, by = c(&quot;chunk&quot;, &quot;chrom&quot;)</code></pre>
<pre class="r"><code>lnz_df_g &lt;- group_by(lnz_df,chrom_chunk,fgeneid) %&gt;% summarize(sum_lnZ=sum(lnZ)) %&gt;% ungroup()

ggplot(lnz_df_g)+geom_point(aes(x=chrom_chunk,y=sum_lnZ,col=factor(fgeneid)))+guides(col=guide_legend(title=&quot;Gene&quot;))+ggtitle(&quot;Sum of variational lower across the genome&quot;,subtitle = &quot;Summation over 5x5 grid of parameter values for each 25k chunk&quot;)</code></pre>
<p><img src="figure/initial_grid_search.Rmd/summ_one_gene-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>It looks like the variational lower bound for part of chromosome 6 is (occasionally) really high. Let’s remove it. (This likely contains the HLA region)</p>
<pre class="r"><code>lnz_df_g &lt;- filter(lnz_df_g,sum_lnZ&lt;0)

ggplot(lnz_df_g)+geom_point(aes(x=chrom_chunk,y=sum_lnZ,col=factor(fgeneid)))+guides(col=guide_legend(title=&quot;Gene&quot;))</code></pre>
<p><img src="figure/initial_grid_search.Rmd/filter_genes-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="parameter-estimates-across-the-genome" class="section level2">
<h2>Parameter estimates across the genome</h2>
<p>Let’s look at values of <span class="math inline">\(\sigma_{\beta,j}\)</span> and <span class="math inline">\(\theta_i\)</span> that give the highest variational lower bounds for each gene</p>
<p><span class="math display">\[\hat{\sigma}_{\beta,p}=\underset{j}{\arg \max} f_{p,g,i,j}(\theta_i,\sigma_{\beta,j},\hat{\beta}_{g,p},\hat{s}_{g,p})  \]</span> <span class="math display">\[\hat{\theta}_p=\underset{i}{\arg \max} f_{p,g,i,j}(\theta_i,\sigma_{\beta,j},\hat{\beta}_g,\hat{s}_g)\]</span></p>
<pre class="r"><code>max_param &lt;- group_by(lnz_df,chrom,chunk,fgeneid) %&gt;% filter(lnZ==max(lnZ))
ggplot(max_param)+  geom_point(aes(x=chrom_chunk,y=sigb,col=factor(fgeneid)))+
  guides(col=guide_legend(title=&quot;Gene&quot;))+
  ggtitle(expression(sigma[beta]),subtitle = &quot;Maximum perchoromosomal chunk&quot;)+
  xlab(&quot;chromosome&quot;)+ylab(expression(sigma[beta]))</code></pre>
<p><img src="figure/initial_grid_search.Rmd/max_est-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(max_param)+geom_point(aes(x=chrom_chunk,y=log10odds,col=factor(fgeneid)))+guides(col=guide_legend(title=&quot;Gene&quot;))+ggtitle(expression(theta),subtitle = &quot;Maximum per chromosomal chunk&quot;)+xlab(&quot;chromosome&quot;)+ylab(expression(theta))</code></pre>
<p><img src="figure/initial_grid_search.Rmd/max_est_2-1.png" width="672" style="display: block; margin: auto;" /></p>
<!-- ## Gene information -->
<!-- ```{r,echo=F} -->
<!-- # fgeneids <- unique(lnz_df$fgeneid) -->
<!-- # eqtlf <- "/media/nwknoblauch/Data/GTEx/GTEx_eQTL/Whole_Blood/SNP.chr1_1_Whole_Blood.h5" -->
<!-- #  -->
<!-- # eqtlleg <- read_df_h5(eqtlf,as.character(fgeneids[1])) -->
<!-- #  -->
<!-- # as_snpleg <- read_df_h5("/media/nwknoblauch/Data/GTEx/GTEx_SNP_h5/Adipose_Subcutaneous.h5","SNPinfo") -->
<!-- # wb_snpleg <- read_df_h5("/media/nwknoblauch/Data/GTEx/GTEx_SNP_h5/Whole_Blood.h5","SNPinfo") -->
<!-- ``` -->
</div>
<div id="session-information" class="section level2">
<h2>Session Information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.3.3 (2017-03-06)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 16.04.2 LTS

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] rssr_0.1-0      lazyeval_0.2.0  h5_0.9.8        RColumbo_0.1.0 
[5] workflowr_0.4.0 rmarkdown_1.3   ggplot2_2.2.1   tidyr_0.6.1    
[9] dplyr_0.5.0    

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.10        knitr_1.15.1        magrittr_1.5       
 [4] munsell_0.4.3       colorspace_1.3-2    R6_2.2.0           
 [7] stringr_1.2.0       plyr_1.8.4          tools_3.3.3        
[10] grid_3.3.3          gtable_0.2.0        DBI_0.6            
[13] git2r_0.18.0        htmltools_0.3.5     RcppParallel_4.3.20
[16] yaml_2.1.14         assertthat_0.1      rprojroot_1.2      
[19] digest_0.6.12       tibble_1.2          rsconnect_0.7      
[22] memoise_1.0.0       RSQLite_1.1-2       evaluate_0.10      
[25] labeling_0.3        stringi_1.1.2       scales_0.4.1       
[28] backports_1.0.5    </code></pre>
</div>

<hr>
<p>
    This <a href="http://rmarkdown.rstudio.com">R Markdown</a> site was created with <a href="https://github.com/jdblischak/workflowr">workflowr</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
